app/services/schools/generate_gas_dashboard_chart_configuration.rb

```diff
@@ -0,0 +1,30 @@
+require 'dashboard'
+
+module Schools
+  class GenerateGasDashboardChartConfiguration
+    def initialize(
+        school,
+        aggregated_meter_collection,
+        fuel_configuration
+      )
+      @school = school
+      @aggregated_meter_collection = aggregated_meter_collection
+      @fuel_configuration = fuel_configuration
+    end
+
+    def generate
+      @school.configuration.update!(gas_dashboard_chart_type: gas_dashboard_chart_type)
+    end
+
+  private
+
+    def gas_dashboard_chart_type
+      return Schools::Configuration::NO_CHART unless @fuel_configuration.has_gas
+      chart_config = { y_axis_units: :kwh }
+      working_chart = Schools::Configuration.gas_dashboard_chart_types.keys.reverse.find do |chart_type|
```

Comment:
might be worth moving `Schools::Configuration.gas_dashboard_chart_types.keys.reverse` out onto a separate line so it can have a variable name that explains what's going on?